src/hotspot/share/utilities/bitCast.hpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2017, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023, Google and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_UTILITIES_BIT_CAST_HPP
+#define SHARE_UTILITIES_BIT_CAST_HPP
+
+// Convert between two types while preserving bit representation.
+
+// To bit_cast<To>(From)
+//
+// Casts from type From to type To without changing the underlying bit representation. This is
+// partially compatible with std::bit_cast introduced in C++20, but is more restrictive on the type
+// of conversions allowed.
```

Comment:
If it's not compatible (or at least trying to be, in so far as possible within the restrictions we operate under)
with `std::bit_cast` (and I've suggested there are reasons why we shouldn't be compatible), then using the
name `bit_cast` seems like a mistake that will only cause confusion.

`PrimitiveConversions::cast` might not be the best name either (John Rose certainly expressed dislike), but
discussing a name change is probably better done outside the context of a PR.

In the meantime, I don't think this part of the change should be made.  There are various places that should
be using PrimitiveConversions::cast (under whatever name is ultimately chosen) but aren't, and they can be
changed to do so.  Later, once we've picked the final name, we can do a global rename.